src/http/parser.rs

```diff
@@ -0,0 +1,290 @@
+use std::collections::{HashMap, LinkedList};
+
+use serde::de::{DeserializeOwned, Error};
+use serde_json::{Map, Value};
+
+use crate::async_graphql_hyper::GraphQLRequestLike;
+use crate::blueprint::Definition;
+use crate::parser::de::{de_kebab, next_token, to_json, to_json_str};
+
+#[derive(Debug, serde::Deserialize, Default)]
+pub struct Parser {
+  operation: String,
+  arguments: Option<String>,
+  selections: Option<String>,
+}
+
+impl Parser {
+  pub fn from_path(path: &str) -> anyhow::Result<Parser> {
+    let path = urlencoding::decode(&de_kebab(path)).unwrap().replace('\\', "");
+    let mut root = String::new();
+    let split = path.split('?').collect::<Vec<&str>>();
```

Comment:
better accept hyper::Uri instead of raw str that already has parsed uri and do not parse it yourself